app/routes.py

```diff
@@ -85,7 +85,20 @@ def settings_profile():
 def settings_preferences():
     return render_template('settings_preferences.html', title='Settings')
 
-@app.route('/settings/password')
+@app.route('/settings/password', methods=['GET', 'POST'])
 @login_required
 def settings_password():
-    return render_template('settings_password.html', title='Settings')
\ No newline at end of file
+    form = PasswordForm()
+    if form.validate_on_submit():
+        user = User.query.filter_by(id=int(current_user.id)).first()
+        if user is None:
+            flash('Error! We could not update your password.', 'error')
+            return redirect(url_for('settings_password'))
+        if not user.check_password(form.currentpassword.data):
+            flash('Error! Your current password appears to be incorrect.', 'error')
+            return redirect(url_for('settings_password'))
+
+        user.set_password(form.newpassword.data)
+        db.session.commit()
+        return redirect(url_for('settings_password'))

```

Comment:
- [ ] Before redirecting, add a flash message to tell the user we successfully updated their password.

```
flash('Your password was updated!', 'info')
```